src/_P144_Vindriktning.ino

```diff
@@ -0,0 +1,379 @@
+      // the plugin settings should be saved to PCONFIG(x)
+      // ping configuration should be read from CONFIG_PIN1 and stored
+      //serialHelper_webformSave(event);
+
+      // after the form has been saved successfuly, set success and break
+      success = true;
+      break;
+    }
+    case PLUGIN_INIT:
+    {
+      // this case defines code to be executed when the plugin is initialised
+      int8_t rxPin = serialHelper_getRxPin(event);
+      int8_t txPin = serialHelper_getTxPin(event);
+      // Try to open the assocaited serial port
+      P144_easySerial = new (std::nothrow) ESPeasySerial(serialHelper_getSerialType(event), rxPin, txPin);
+      if (P144_easySerial != nullptr) 
+      {
+        P144_easySerial->begin(9600);
+        success = true;
+      }
+
+      #ifdef PLUGIN_144_DEBUG
+      if (loglevelActiveFor(LOG_LEVEL_INFO))
+      {
+        String log = F("P144 : Init: ");
+        if (success)
+        {
+          log += F("  ESP GPIO-pin RX:");
+          log += rxPin;
+          log += F(" TX:");
+          log += txPin;  
+        }
+        else
+        {
+          log += F("Failed opening serial port");
+        }
+        addLogMove(LOG_LEVEL_INFO, log);
+      }
+      #endif
+      break;
+    }
+
+    case PLUGIN_READ:
+    {
+      // code to be executed to read data
+      // It is executed according to the delay configured on the device configuration page, only once
+      UserVar[event->BaseVarIndex] = P144_pm25;
+      #ifdef PLUGIN_144_DEBUG
+      if (loglevelActiveFor(LOG_LEVEL_INFO))
+      {
+        String log = F("P144 : READ ");
+        log += P144_pm25;
+        addLogMove(LOG_LEVEL_INFO, log);
+      }
+      #endif
+      // after the plugin has read data successfuly, set success and break
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WRITE:
+    {
+      // this case defines code to be executed when the plugin executes an action (command).
+      // Commands can be accessed via rules or via http.
+      // As an example, http://192.168.1.12//control?cmd=dothis
+      // implies that there exists the comamnd "dothis"
+
+      break;
+    }
+
+    case PLUGIN_EXIT:
+    {
+      // perform cleanup tasks here. For example, free memory
+      if (P144_easySerial != nullptr) {
+        delete(P144_easySerial);
+        P144_easySerial = nullptr;
+      }
+      success = true;
+      break;
+    }
+
+    case PLUGIN_ONCE_A_SECOND:
+    {
+      // code to be executed once a second. Tasks which do not require fast response can be added here
+      success = true;
+      break;
+    }
+
+    case PLUGIN_TEN_PER_SECOND:
+    {
+      // code to be executed 10 times per second. Tasks which require fast response can be added here
+      // be careful on what is added here. Heavy processing will result in slowing the module down!
+      bool new_data = false;
+      while ((P144_easySerial->available() > 0) && !new_data) 
```

Comment:
Just one remark about using the `available()` function of the serial port.
Can you fetch this value once and then loop over it with `read()` ?
The reason is that if one might be using this on another UART port like the I2C to UART bridge we also support, then this call to `available()` is just as "expensive" as performing a read.
So you're essentially crippling the throughput of that serial port.

Given you run this in the PLUGIN_TEN_PER_SECOND call, only looping once over the reported nr from `available()` is enough here.

Reply:
Thanks for the remark, I was not aware of this. I will be on a business trip till coming saturday, so changes will be later.